lib/plug/conn.ex

```diff
@@ -1356,6 +1373,27 @@ end
 
 defimpl Collectable, for: Plug.Conn do
   def into(conn) do
+    IO.puts(:stderr, """
+    warning: using Enum.into/2 for conn is deprecated, use Enum.reduce_while/3 instead:
+
+    To stream data use `Enum.reduce_while/3` instead of `Enum.into/2`.
+    `Enum.reduce_while/3` allows aborting the execution if `chunk/2` fails to
+    deliver the chunk of data.
+
+    Example
+
+        ~w(each chunk as a word)
+        |> Enum.reduce_while(conn, fn (chunk, conn) ->
+          case Plug.Conn.chunk(conn, chunk) do
+            {:ok, conn} ->
+              {:cont, conn}
+            {:error, :closed} ->
+              {:halt, conn}
+          end
+        end)
+
+    """)
+
```

Comment:
If we cant get around runtime warnings we should try to keep them short and point to documentation instead to reduce pressure and spam in the logging system.

Sometimes runtime warnings can even be breaking changes because they make your logging useless or at worst breaks it due to pressure.

Reply:
@ericmj good call. Done in master.